mne/preprocessing/_csd.py

```diff
@@ -148,8 +148,9 @@ def compute_current_source_density(inst, picks=None, lambda2=1e-5,
                              H=H, radius=radius)
 
     if isinstance(inst, BaseEpochs):
-        inst._data = np.array([np.dot(trans_csd, epo)
-                               for epo in inst._data[:, picks]])
+        inst._data = inst._data[:, picks]
+        for epo in inst._data:
+            epo = np.dot(trans_csd, epo)
```

Comment:
You need to do `epo[:] = np.dot(trans_csd, epo)`, otherwise the correct bit of data is not overwritten.

But paired with the removal of the `inst._data` line above, you'll want:
```
epo[picks] = np.dot(trans_csd, epo[picks])
```

Reply:
Hmmm, the way I wrote it still passed the tests. I changed it but that's not good. Is there something wrong with the tests?

Reply:
So the way Dennis wrote the tests before had it so that if you only pick 4 of the six channels for instance, the epochs object you get back only has the ones that you pick: i.e. that would suggest you would want to toss all channels that you don't transform. What do you think should be done?

Reply:
I would transform all EEG channels. If people don't want them all transformed they can drop using drop_channels beforehand. Better separation of functionality